### ISSUE TITLE
pybind11Tools.cmake - unknown argument specified

### ISSUE THREAD

**Issue by gitartpiano** [pybind/pybind11#4325]
The symptom:
```
CMake Error at vendor/pybind11/tools/pybind11Tools.cmake:217 (if):
  if given arguments:
    "NOT" "MSVC" "AND" "NOT" "TEST" "MATCHES" "DEBUG|RELWITHDEBINFO"
  Unknown arguments specified
```

The fix I have found:
Was:
```
if(NOT MSVC AND NOT ${uppercase_CMAKE_BUILD_TYPE} MATCHES DEBUG|RELWITHDEBINFO)
```

Changed to:
```
if(NOT MSVC AND NOT "${uppercase_CMAKE_BUILD_TYPE}" MATCHES DEBUG|RELWITHDEBINFO)
```
At:
https://github.com/pybind/pybind11/blob/ee2b5226295d67b690faddd446a329bb2840a1a8/tools/pybind11Tools.cmake#L217

Please consider applying this fix, or suggesting a better solution to above error.

**Comment by gitartpiano** [comment]
The version I have found the problem: v2.10.1

**Comment by Skylion007** [comment]
Closed by #4327
